app/services/moab_validation_handler.rb

```diff
@@ -90,4 +94,16 @@ def set_status_as_seen_on_disk(found_expected_version)
       update_status('validity_unknown')
     end
   end
+
+  def handle_missing_moab
+    transaction_ok = ActiveRecordUtils.with_transaction_and_rescue(results) do
+      results.add_result(AuditResults::MOAB_NOT_FOUND,
+                         db_created_at: complete_moab.created_at.iso8601,
+                         db_updated_at: complete_moab.updated_at.iso8601)
```

Comment:
The reason that I moved it into the transaction is because (1) if we're going to expect handling of RecordNotFound and (2) the first call to complete_moab is indeterminate, then all uses of complete_moab must be wrapped in a transaction.

Gotta say, I'm not thrilled with this pattern.

Reply:
ah!  yeah, i missed that.  makes sense.

sorry again for introducing this module, i am obviously also finding it unintuitive to reason about when coming back to it.